os/arch/arm/src/armv7-m/up_assert.c

```diff
@@ -151,6 +151,25 @@ static void up_stackdump(uint32_t sp, uint32_t stack_base)
 
 	for (stack = sp & ~0x1f; stack < stack_base; stack += 32) {
 		uint32_t *ptr = (uint32_t *)stack;
+		uint32_t start_addr = stack;
+
+		if (start_addr < sp) {
```

Comment:
You could this at out of for loop.
1. check (sp & ~0x1f) ~ (sp +@) (To be 8 values)
2. check (sp + @) ~ stack base
```
	for (stack = sp & ~0x1f; stack < sp; stack += 32) {
                  // print xxxxx for before of sp and real values for next of sp
                  // you could use "stack" itself and then you can start next for loop with "stack" variable.
	}
	for (stack = last_addr_of_previous_loop; stack < stack_base; stack += 32) {
		uint32_t *ptr = (uint32_t *)stack;
		lldbg("%08x: %08x %08x %08x %08x %08x %08x %08x %08x\n",
			   stack, ptr[0], ptr[1], ptr[2], ptr[3], ptr[4], ptr[5], ptr[6], ptr[7]);
	}
```

Reply:
But this is not important you could select the way.